python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -203,6 +203,13 @@ class NearestNeighbors(Base):
         self.devices = devices
         self.n_neighbors = n_neighbors
         self._should_downcast = should_downcast
+        self.sizes = None
+        self.inputs = None
+
+    def __del__(self):
+        if self.sizes is not None:
+            free(<int*><size_t>self.sizes)
```

Comment:
I'm surprised that you need the <size_t> cast first when doing int* too...

Reply:
This is a cython thing, unfortunately. `size_t` is implicitly convertable to between cython and python.

Reply:
This trick actually turns out to be quite useful, btw. It allows us to encode the pointer so that we can pass it between objects at runtime. This is the cython equivalent of void*.